### PR TITLE
Update Arc helm chart with pre-install hook and Gatekeeper beta8 release

### PR DESCRIPTION
**Pull request by RamyasreeChakka**
Update Arc helm chart with pre-install hook and Gatekeeper beta8 release.

PR changes:
1. Gatekeeper image changed to latest release beta8
2. Remove CaBundle from gatekeeper.yaml so that cert generation is not messed up with reconciliation logic using helm upgrade.
3. Pre-install hook added to patch kube-system and azure-arc namespaces with control-plane label.

Testing:
- Verified the golden set test scenarios shared in email
- Verified patching label in below scenarios. Using strategic merge so that new label is appended to existing list if any.
1.      Namespaces do not have any labels.
2.      Namespace already has control-plane label.
3.      Namespace has some labels other than control-plane.
4.      Verified patch job, service account, roles are deleted on hook completion with success.

### CODE REVIEW

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+subjects:
+- kind: ServiceAccount
+  name: azure-policy-patch-label-account
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: azure-policy-patch-label-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: RoleBinding
```
Comment (EmandM) [inline extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml:50]: Why is this RoleBinding duplicated?

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        - name: azure-policy-patch-label
+          image: {{ .Values.alpine.image.name }}:{{ .Values.alpine.image.tag }}
+          imagePullPolicy: IfNotPresent
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              set -o pipefail
+              token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+              namespaces="kube-system azure-arc"
```
Comment (EmandM) [inline extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml:97]: This should maybe be an environment variable? Makes it easier to see and change

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+              for namespace in ${namespaces}
+              do
+                echo "Retrieving namespace ${namespace} details"
+                namespaceDetails=$(curl -s -X GET -k https://kubernetes.default.svc/api/v1/namespaces/${namespace} \
+                -H "Authorization: Bearer ${token}" \
+                -H "Content-Type: application/json" \
+                -H "Accept: application/json")
+                kind=$(echo $namespaceDetails | jq .kind)
+                if [ "$kind" = "\"Namespace\"" ]
```
Comment (EmandM) [inline extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml:106]: I assume this check is to ensure an error resource isn't returned. Is this correct?

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml (hunk 1 above)
Comment (RamyasreeChakka) [reply extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml:50]: Azure policy patch label service account needs get/patch permissions on only two namespaces -> kube-system and azure-arc. The cluster role defines the namespace permissions get/patch. The role bindings does the assignment for service account in azure-arc and kube-system namespace.

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml (hunk 3 above)
Comment (RamyasreeChakka) [reply extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml:106]: Yes, that's right.

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml (hunk 2 above)
Comment (RamyasreeChakka) [reply extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml:97]: For now its fine I think as this is a short-term solution anyways.

File: extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml (hunk 1 above)
Comment (EmandM) [reply extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml:50]: Ah, that makes sense

**Pull request merged**